oep_framework/oep/BNBOffscreenEffectPlayer.mm

```diff
@@ -39,6 +40,10 @@ - (instancetype)initWithWidth:(NSUInteger)width
     m_ep = bnb::oep::effect_player::create(paths, std::string([token UTF8String]));
     m_ort = std::make_shared<bnb::offscreen_render_target>();
     m_oep = bnb::oep::interfaces::offscreen_effect_player::create(m_ep, m_ort, width, height);
+    
+    auto me_ort = std::dynamic_pointer_cast<bnb::oep::interfaces::offscreen_render_target_metal_extension>(m_ort);
```

Comment:
https://en.cppreference.com/w/cpp/memory/shared_ptr/pointer_cast
"If r is empty, so is the new shared_ptr (but its stored pointer is not necessarily null). Otherwise, the new shared_ptr will share ownership with the initial value of r, except that it is empty if the dynamic_cast performed by dynamic_pointer_cast returns a null pointer."

I think something was missed and can cause the crash in some cases.